pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -104,6 +104,11 @@ func NewCodec(
 			}
 			internal.encodeVersion[v.Group] = v
 		}
+		if len(internal.encodeVersion) == 1 {
+			for _, v := range internal.encodeVersion {
```

Comment:
I don't understand it - why do you want to have a for if there is exactly one element in the slice?